tools/clang/unittests/HLSL/PixTest.cpp

```diff
@@ -3269,6 +3278,168 @@ void main()
   VERIFY_IS_TRUE(FoundTheVariable);
 }
 
+class DxcIncludeHandlerForInjectedSourcesForPix : public IDxcIncludeHandler {
+private:
+  DXC_MICROCOM_REF_FIELD(m_dwRef)
+
+  std::vector<std::pair<std::wstring, std::string>> m_files;
+  PixTest *m_pixTest;
+
+public:
+  DXC_MICROCOM_ADDREF_RELEASE_IMPL(m_dwRef)
+  DxcIncludeHandlerForInjectedSourcesForPix(
+      PixTest *pixTest, std::vector<std::pair<std::wstring, std::string>> files)
+      : m_dwRef(0), m_pixTest(pixTest), m_files(files){};
+
+  HRESULT STDMETHODCALLTYPE QueryInterface(REFIID iid, void **ppvObject) {
+    return DoBasicQueryInterface<IDxcIncludeHandler>(this, iid, ppvObject);
+  }
+
+  HRESULT insertIncludeFile(_In_ LPCWSTR pFilename,
+                            _In_ IDxcBlobEncoding *pBlob, _In_ UINT32 dataLen) {
+    return E_FAIL;
+  }
+
+  HRESULT STDMETHODCALLTYPE LoadSource(
+      _In_ LPCWSTR pFilename,
+      _COM_Outptr_result_maybenull_ IDxcBlob **ppIncludeSource) override {
```

Comment:
I thought we want to remove these _In_  and _COM_Outptr_result_maybenull_?

Reply:
ah yes! I have removed all the SAL